File: 2024-2/sesiones-de-clase/sesion-de-clase-21/Alumno.cpp
/* 
 * File:   Alumno.cpp
 * Author: erichuiza
 * 
 * Created on November 11, 2024, 10:32 AM
 */

#include "Alumno.h"

Alumno::Alumno() {
    codigo = {};
    nombre = {};
    apellido = {};
    edad = {};
    ciclo = {};
    promedio = {};
}

Alumno::Alumno(const Alumno& orig) {
    *this = orig;
}

int Alumno::getCodigo() const {
    return codigo;
}

double Alumno::getPromedio() const {
    return promedio;
}

string Alumno::getNombre() const {
    return nombre;
}

string Alumno::getApellido() const {
    return apellido;
}

Alumno& Alumno::operator=(const Alumno& orig) {
    codigo = orig.codigo;
    nombre = orig.nombre;
    apellido = orig.apellido;
    edad = orig.edad;
    ciclo = orig.ciclo;
    promedio = orig.promedio;
    
    return *this;
}

bool Alumno::operator<(const Alumno& alumno) const {
    return promedio < alumno.promedio;
}

bool Alumno::operator>(const Alumno& alumno) const {
    return promedio > alumno.promedio;
}

void Alumno::cargar(ifstream& archivo) {
    archivo >> codigo;
    archivo.ignore();
    getline(archivo, nombre, ',');
    getline(archivo, apellido, ',');
    archivo >> edad;
    archivo.ignore();
    archivo >> ciclo;
    archivo.ignore();
    archivo >> promedio;
    archivo.ignore();
}

void Alumno::imprimir(ofstream& archivo) const {
    archivo << setw(10) << codigo << setw(20) << nombre << setw(20) << apellido << setw(5) << edad << setw(5) << ciclo << setw(5) << promedio;
}

ifstream& operator>>(ifstream& archivo, Alumno& alumno) {
    alumno.cargar(archivo);
    return archivo;
}

ofstream& operator<<(ofstream& archivo, const Alumno& alumno) {
    alumno.imprimir(archivo);
    return archivo;
}

Alumno::~Alumno() {
}

File: 2024-2/sesiones-de-clase/sesion-de-clase-25/src/Alumno.java

import java.util.Scanner;

/**
 *
 * @author erichuiza
 */
abstract class Alumno {
    private char tipo;
    private int codigo;
    private String nombre;
    private String apellido;
    private int edad;
    private int ciclo;
    private double promedio;
    
    public Alumno() {
        System.out.println("Constructor Alumno.");
    }
    
    public Alumno(char tipo) {
        this.tipo = tipo;
    }
    
    public void setTipo(char tipo) {
        this.tipo = tipo;
    }

    public void setCodigo(int codigo) {
        this.codigo = codigo;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public void setApellido(String apellido) {
        this.apellido = apellido;
    }

    public void setEdad(int edad) {
        this.edad = edad;
    }

    public void setCiclo(int ciclo) {
        this.ciclo = ciclo;
    }

    public void setPromedio(double promedio) {
        this.promedio = promedio;
    }

    public char getTipo() {
        return tipo;
    }

    public int getCodigo() {
        return codigo;
    }

    public String getNombre() {
        return nombre;
    }

    public String getApellido() {
        return apellido;
    }

    public int getEdad() {
        return edad;
    }

    public int getCiclo() {
        return ciclo;
    }

    public double getPromedio() {
        return promedio;
    }
    
    public void cargar(Scanner scanner) {
        codigo = scanner.nextInt();
        nombre = scanner.next();
        apellido = scanner.next();
        edad = scanner.nextInt();
        ciclo = scanner.nextInt();
        promedio = scanner.nextDouble();
    }
    
    public void imprimir() {
        System.out.println("Tipo: " + tipo);
        System.out.println("Codigo: " + codigo);
        System.out.println("Nombre: " + nombre);
        System.out.println("Apellido: " + apellido);
        System.out.println("Edad: " + edad);
        System.out.println("Ciclo: " + ciclo);
        System.out.println("Promedio: " + promedio);
    }
}
